uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -176,27 +195,42 @@ impl APIConverter<Enum> for weedle::InterfaceDefinition<'_> {
 /// Represents an individual variant in an Enum.
 ///
 /// Each variant has a name and zero or more fields.
+
+#[derive(Debug)]
+pub struct Variant<'a, Parent: CINode> {
+    pub(super) parent: &'a Parent,
```

Comment:
So this starts to get a bit complicated...the `Variant` struct may be used as part of an `Enum` or as part of an `Error`, so its parent pointer type needs to be generic. The main property that we want from the parent pointer is the ability to get all the way up to the containing `ComponentInterface`, so I made a trait for that.

An alternative could be to have it point directly to the containing `ComponentInterface` rather than its containing object, which would make this more of an "owner" pointer than a "parent" pointer. But, I can imagine reasons why a variant might need to be able to inspect the properties of its containing Enum/Error.